js/canvas.ts

```diff
@@ -160,21 +123,30 @@ export class LoresPage2D implements LoresPage {
 
     private _buffer: memory[] = [];
     private _refreshing = false;
-    private _monoMode = false;
     private _blink = false;
 
+    private highColorTextMode = false
+
     dirty: Region = {...notDirty}
     imageData: ImageData;
 
-    constructor(private page: number,
+    constructor(
+        private vm: VideoModes,
+        private page: pageNo,
         private readonly charset: rom,
-        private readonly e: boolean) {
-        this.imageData = new ImageData(560, 192);
+        private readonly e: boolean
+    ) {
+        if (!tmpContext) {
+            throw new Error('No 2d context');
+        }
+        this.imageData = tmpContext.createImageData(560, 192);
         for (let idx = 0; idx < 560 * 192 * 4; idx++) {
```

Comment:
Extract a method for this?

Reply:
I think I can just use `fill()` these days.